letter-shop/index.md

```diff
@@ -0,0 +1,634 @@
+To do this we need a **server**, run by the shop, and also a **client**, run by the customer. 
+
+### Server
+
+The server will perform two functions:
+1. It will host an HTTP service where clients can GET a new random letter with each request.
+1. It will listen for incoming ILP payments related to pending letter purchases, fulfill these to accept the payment, and provide the payer with a token they can use to claim their letter.
+
+Once we have completed the tutorial the server will perform the following steps:
+
+1. Connect to an account and monitor it for incoming payments
+1. Start a web server to accept letter requests
+1. Process a request, store the result, and provide the customer with details of how to pay for the request and get the result
+1. Process an incoming payment related to a previous request, and provide payer with details of how to perform request again using proof-of-payment token
+1. Process a request with an attached proof-of-payment token, and return the originally generated result
+
+### Client
+
+The client will perform an ILP payment and output the returned fulfillment. This can be redeemed to get the letter that was requested in the original request.
+
+Once we have completed the tutorial the client will perform the following steps:
+
+1. Connect to an account from which to send payments
+1. Initiate a payment using the address, amount, and condition provided in the response to the original letter request
+1. Complete the payment and return the proof-of-payment token to be used to complete the original request
+
+## Step 1: Get the code
+
+The code for this tutorial is available in two forms:
+
+1. If you follow the tutorial you will start with some very basic code and will flesh it out as we go. 
+2. If you'd prefer to start with the finished code and simply walk through the steps then you can do that too.
+
+First let's get all the code, either using Git or by simply downloading and unzipping it.
+
+#### Git
+
+```shell
+git clone https://github.com/interledger/tutorials
+```
+
+#### Download
+
+[ZIP Archive](https://github.com/interledger/tutorials/archive/master.zip)
+
+Assuming you have either cloned the repository, or unzipped your download, open a terminal window at the root of the project and change into the `letter-shop` directory.
+
+You should see the following files:
+
+```shell
+README.md
+package.json
+pay.js
+plugins.js
+proxy.js
+shop.js
+```
+
+Most of these files are just scaffolding, waiting for you to complete as we go. If you want the complete files so you can just follow along, switch into the `complete` sub-directory.
+
+## Step 2: Run the server
+
+The first thing we need to do is install our dependencies and start our shop's server.
+
+We install using npm (this may take a few minutes while it downloads all the dependencies):
+
+```shell
+npm install
+```
+
+> If you get an error here make sure you have `node` installed.
+
+Then we try and start our shop server:
+
+```shell
+node shop.js
+```
```

Comment:
I love the "No account configured yet for the Shop. See 'plugins.js'." thing! Makes it really interactive.